public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -1,19 +1,133 @@
+import {
+  RulerRulesConfigDTO,
+  RulerRuleGroupDTO,
+  RulerRuleDTO,
+  RulerGrafanaRuleDTO,
+  RulerAlertingRuleDTO,
+} from 'app/types/unified-alerting-dto';
 
 import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
 import { updateLotexNamespaceAndGroupAction } from '../../state/actions';
 import { checkEvaluationIntervalGlobalLimit } from '../../utils/config';
 import { getRulesSourceName } from '../../utils/datasource';
 import { initialAsyncRequestState } from '../../utils/redux';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
 import { EvaluationIntervalLimitExceeded } from '../InvalidIntervalWarning';
 import { evaluateEveryValidationOptions } from '../rule-editor/GrafanaEvaluationBehavior';
 
+const MINUTE = '1m';
+interface AlertWithFor {
+  alertName: string;
+  forDuration: string;
+}
+
+export const isRulerGrafanaRuleDTO = (rule: RulerRuleDTO): rule is RulerGrafanaRuleDTO => 'grafana_alert' in rule;
+export const isAlertingRuleDTO = (rule: RulerRuleDTO): rule is RulerAlertingRuleDTO => 'alert' in rule;
+
+export const getAlertInfo = (alert: RulerRuleDTO): AlertWithFor => {
+  const emptyAlert: AlertWithFor = {
+    alertName: '',
+    forDuration: '0s',
+  };
+  if (isRulerGrafanaRuleDTO(alert)) {
+    return {
+      alertName: alert.grafana_alert.title,
+      forDuration: alert.for,
+    };
+  }
+  if (isAlertingRuleDTO(alert)) {
+    return {
+      alertName: alert.alert,
+      forDuration: alert.for ?? '1m',
+    };
+  }
+  return emptyAlert;
+};
+export const getIntervalForGroup = (
+  rulerRules: RulerRulesConfigDTO | null | undefined,
+  group: string,
+  folder: string
+) => {
+  const folderObj: Array<RulerRuleGroupDTO<RulerRuleDTO>> = rulerRules ? rulerRules[folder] : [];
+  const groupObj = folderObj?.find((rule) => rule.name === group);
+
+  const interval = groupObj?.interval ?? MINUTE;
+  return interval;
+};
+
+// parseDuration method needs units separated by space
+export const safeParseDurationstr = (duration: string) => {
+  const reg = /(?<=[a-z])(?=\d)/i;
+  return parseDuration(duration.replace(reg, ' '));
+};
+
+type AlertsWithForTableColumnProps = DynamicTableColumnProps<AlertWithFor>;
+type AlertsWithForTableProps = DynamicTableItemProps<AlertWithFor>;
+
+export const RulesForGroupTable = ({
+  rulerRules,
+  group,
+  folder,
+}: {
+  rulerRules: RulerRulesConfigDTO | null | undefined;
+  group: string;
+  folder: string;
+}) => {
+  const styles = useStyles2(getStyles);
+  const folderObj: Array<RulerRuleGroupDTO<RulerRuleDTO>> = rulerRules ? rulerRules[folder] : [];
+  const groupObj = folderObj?.find((rule) => rule.name === group);
+  const rules: RulerRuleDTO[] = groupObj?.rules ?? [];
+
+  const rows: AlertsWithForTableProps[] = rules
+    .slice()
+    .sort(
+      (alert1, alert2) =>
+        durationToMilliseconds(safeParseDurationstr(getAlertInfo(alert1).forDuration)) -
```

Comment:
Could you try to use `parsePrometheusDuration` for parsing? I think it works best with Prometheus durations

Reply:
Done! And you're right, this `parsePrometheusDuration` method simplifies the code.